pkg/kapp/resourcesmisc/custom_waiting_resource.go

```diff
@@ -58,6 +59,10 @@ func (s CustomWaitingResource) IsDoneApplying() DoneApplyState {
 	for _, condMatcher := range s.waitRule.ConditionMatchers {
 		for _, cond := range obj.Status.Conditions {
 			if cond.Type == condMatcher.Type && cond.Status == condMatcher.Status {
+				if condMatcher.SupportsObservedGeneration && obj.Metadata.Generation != cond.ObservedGeneration {
```

Comment:
multiple conditions might be in play. some of them might be reporting old health (ie have old observedGeneration). i think we would want to ignore conditions that we "support observed generation for", but also it does not have current generation. if we just return then we would exit too quick (especially if there are other conditions that may be considered updated).

Reply:
@cppforlife  updated the logic to check each condition instead of returning early.